config/initializers/ldap_instrumentation.rb

```diff
@@ -1,45 +1,5 @@
 # Debug logging from net-ldap and ldap_fluff events sent via ActiveSupport::Notifications
-module Foreman
-  class LdapSubscriber < ActiveSupport::LogSubscriber
-    include Foreman::TelemetryHelper
-
-    def logger
-      ::Foreman::Logging.logger('ldap')
-    end
-
-    def self.define_log(action, log_name, color)
-      define_method(action) do |event|
-        telemetry_observe_histogram(:ldap_request_duration, event.duration)
-        return unless logger.debug?
-        name = '%s (%.1fms)' % [log_name, event.duration]
-        debug "  #{color(name, color, true)}  [ #{yield(event.payload)} ]"
-      end
-    end
-  end
-
-  class NetLdapSubscriber < LdapSubscriber
-    define_log(:bind, 'op bind', YELLOW) do |payload|
-      if payload[:bind].try(:status)
-        "result=#{payload[:bind].status}"
-      else
-        "result=#{payload[:bind]}"
-      end
-    end
-    define_log(:search, 'op search', YELLOW) { |payload| "filter=#{payload[:filter]}, base=#{payload[:base]}" }
-  end
-
-  class LdapFluffSubscriber < LdapSubscriber
-    define_log(:authenticate, 'authenticate', GREEN) { |payload| "user=#{payload[:uid]}" }
-    define_log(:test, 'test', GREEN) {}
-    define_log(:user_list, 'user_list', GREEN) { |payload| "group=#{payload[:gid]}" }
-    define_log(:valid_group?, 'valid_group?', GREEN) { |payload| "group=#{payload[:gid]}" }
-    define_log(:find_group, 'find_group', GREEN) { |payload| "group=#{payload[:gid]}" }
-    define_log(:group_list, 'group_list', GREEN) { |payload| "user=#{payload[:uid]}" }
-    define_log(:valid_user?, 'valid_user?', GREEN) { |payload| "user=#{payload[:uid]}" }
-    define_log(:find_user, 'find_user', GREEN) { |payload| "user=#{payload[:uid]}" }
-    define_log(:is_in_groups?, 'is_in_groups?', GREEN) { |payload| "user=#{payload[:uid]}, grouplist=#{payload[:grouplist].inspect}" }
-  end
+Rails.application.config.after_initialize do
+  Foreman::Ldap::NetLdapSubscriber.attach_to :net_ldap
```

Comment:
I've just realized this attachment will not reload so the subscribtions would be for the old class after reload.
Can we move the Subscriber classes into a lib folder and require them so those would not reload?

Reply:
```
renamed:    app/services/foreman/ldap/ldap_fluff_subscriber.rb -> lib/foreman/ldap/ldap_fluff_subscriber.rb
renamed:    app/services/foreman/ldap/ldap_subscriber.rb -> lib/foreman/ldap/ldap_subscriber.rb
renamed:    app/services/foreman/ldap/net_ldap_subscriber.rb -> lib/foreman/ldap/net_ldap_subscriber.rb
```